validator/db/kv/migration_snappy_history.go

```diff
@@ -0,0 +1,34 @@
+package kv
+
+import (
+	"bytes"
+
+	"github.com/golang/snappy"
+	bolt "go.etcd.io/bbolt"
+)
+
+var migrationSnappyAttestationHistory0Key = []byte("snappy_attestation_history_0")
+
+
+// Migrate the attestation history data to use snappy compression on disk. This paradigm will
+// significantly reduce disk I/O at the cost of a slightly increased CPU usage. Early benchmarks and
+// tests indicate that this compression saves 25% on disk I/O and storage.
+func migrateSnappyAttestationHistory(tx *bolt.Tx) error {
+	mb := tx.Bucket(migrationsBucket)
+	if b := mb.Get(migrationSnappyAttestationHistory0Key); bytes.Equal(b, migrationCompleted) {
+		return nil // Migration already completed.
+	}
+
+	bkt := tx.Bucket(historicAttestationsBucket)
+
+	// Compress all attestation history data.
+	if err := bkt.ForEach(func(k, v []byte) error {
+		enc := snappy.Encode(nil /*dst*/, v)
```

Comment:
Add ctx error check here as well?

```go
if ctx.Err() != nil {
			return ctx.Err()
		}
```

Reply:
I am not sure this is a great idea. When would we set a deadline or cancel a migration?

Reply:
Probably never 🤔  Scrap this